src/App.tsx

```diff
@@ -15,48 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+interface FilterParams {
+  sortField: SortField;
+  sortReverse: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, sortReverse }: FilterParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortField.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortField.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (sortReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortField.NONE);
+  const [sortReverse, setSortReverse] = useState(false);
```

Comment:
Rename boolean variables according [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)
```suggestion
  const [isReverse, setIsReverse] = useState(false);
```